docs/resource_sets.rst

```diff
@@ -0,0 +1,86 @@
+**********************************
+Resource sets and partial compile
+**********************************
+To reduce the time of a compile action, resource sets can be used. Without the use of those, each time a resource is added,
+updated or removed, the entire model is recompiled.
+By using resource sets, the resources that are usually updated together can be grouped in resource sets.
+This way, the time to compile will depend on the size of the targeted resource set. When not using resource sets,
+the time to compile will depend on the size of the entire model.
+This allows to perform certain actions only on the resources of a specific resource set.
+Compiling only some resource sets and not the entire model is called a partial compile.
```

Comment:
I decided to start fresh for this block of text, also taking into account the other comments on it. I trimmed it down to three sentences that I believe still convey the point clearly. I also added a section to add a bit more context on how to use them (i.e. the compiler does not do any magic, the user needs to modify the model) and a note that this is generally meant to be used through tooling rather than for manual exports.

Reply:
(haven't pushed yet, will notify you when)

Reply:
I will fix the sentencing and structure of it. I think it is better if you address the more technical side of the doc, wherever possible.

Reply:
I felt the two went sort of hand in hand on this one. Of course, feel free to change what I wrote.